InstaScore/Controller/ResultViewController.swift

```diff
@@ -1,91 +1,121 @@
+        var homeLabel = ""
+        var scoreLabel = ""
+        var awayLabel = ""
+        let matchData = matchesGrouped[indexPath.section][indexPath.row]
         
+        homeLabel = matchData.match_hometeam_name
+        scoreLabel = "\(matchData.match_hometeam_score)-\(matchData.match_awayteam_score)"
+        awayLabel = matchData.match_awayteam_name
+        cell.homeTeamLabel.text = homeLabel
+        cell.scoreLabel.text = scoreLabel
+        cell.awayTeamLabel.text = awayLabel
         
-        let cell = tableView.dequeueReusableCell(withIdentifier: "ReusableCell", for: indexPath) as! MatchCell
-        let match = matches[indexPath.row]
-        cell.homeTeamLabel.text = match.homeTeam
-        cell.scoreLabel.text = match.score
-        cell.awayTeamLabel.text = match.awayTeam
+        if matchData.match_status == "" { // not played
+            cell.timeLabel.text = "\(matchData.match_date) \n \(matchData.match_time)"
+        }
+        else { // finished or live
+            if matchData.match_live == "0" { // finished
+                cell.timeLabel.text = matchData.match_status
+            } else if matchData.match_live == "1" {
+                cell.timeLabel.text = "LIVE"
+            }
+            else {
+                cell.timeLabel.text = "ERROR"
+            }
+        }
         return cell
     }
-    
-    
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        view.backgroundColor = .gray
+        let header = UIView(frame: CGRect(x: 0, y: 0, width: view.frame.size.width, height: 50))
+        header.backgroundColor = .black
+        let label = UILabel(frame: CGRect(x: header.frame.minX, y: header.frame.minY, width: header.frame.size.width, height: header.frame.size.height))
+        header.addSubview(label)
+        label.text = matchesGrouped[section][0].league_name
+        label.textAlignment = .center
+        label.textColor = .white
+        return header
+    }
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 50.0
+    }
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return matchesGrouped[section][0].league_name
+    }
 }
-
 extension ResultViewController: UITableViewDelegate{
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        print(indexPath.row)
-        performSegue(withIdentifier: "goToDetails", sender: self)
-        scoreManager.fetchScore(date1: date1, date2: date2)
-          
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let detailsVC =  storyboard.instantiateViewController(withIdentifier: "DetailsStoryBoard") as! DetailsViewController
+        detailsVC.passDataModel.indexChosen = indexPath.row
+        detailsVC.passDataModel.sectionChosen = indexPath.section
+        detailsVC.passDataModel.matchesGrouped = matchesGrouped
```

Comment:
Czemu chcesz przerzucić takie trzy parametry, zamiast przerzucić do DetailViewControllera po prostu jeden mecz?

```suggestion
        detailsVC.passDataModel.match = matchesGrouped[indexPath.section][indexPath.row]
```